polygeist/test/polygeist-opt/selecti1ext.mlir

```diff
@@ -0,0 +1,28 @@
+// RUN: polygeist-opt --canonicalize %s | FileCheck %s 
+  
+// CHECK-LABEL: func.func @test1(%arg0: i1) -> i32 {
+// CHECK:         %0 = arith.select %arg0, %c1_i32, %c2_i32 : i32
+// CHECK-NEXT:    return %0 : i32
+// CHECK-NEXT:  }
+
+func.func @test1(%arg0: i1) -> i32 {
+  %c1_i32 = arith.constant 1 : i32
+  %c2_i32 = arith.constant 2 : i32
+  %0 = arith.select %arg0, %c1_i32, %c2_i32 : i32
+  return %0 : i32
+}
+
+// CHECK-LABEL: func.func @test2(%arg0: i1) -> i32 {
+// CHECK:         %0 = arith.xori %arg0, %true : i1
+// CHECK-NEXT:    %1 = arith.extui %0 : i1 to i32
+// CHECK-NEXT:    return %1 : i32
+// CHECK-NEXT:  }
+
+func.func @test2(%arg0: i1) -> i32 {
+  %true = arith.constant true
+  %false = arith.constant false
+  %0 = arith.extui %true : i1 to i32
+  %1 = arith.extui %false : i1 to i32
+  %2 = arith.select %arg0, %1, %0 : i32
+  return %2 : i32
+}
```

Comment:
Our current pattern would still match this. I guess there's a pattern with a higher benefit/added before. I think we should check that the first and second arguments are [extended] constants 0 and 1 respectively.

Reply:
`SelectI1Ext` is matched first, the simplified by other patterns.
```
Trying to match "SelectI1Ext"
"SelectI1Ext" result 1
// *** IR Dump After Pattern Application ***
func.func @test2(%arg0: i1) -> i32 {
  %c0_i32 = arith.constant 0 : i32
  %c1_i32 = arith.constant 1 : i32
  %false = arith.constant false
  %true = arith.constant true
  %0 = arith.select %arg0, %false, %true : i1
  %1 = arith.extui %0 : i1 to i32
  return %1 : i32
}
Trying to match "SelectI1Simplify"
"SelectI1Simplify" result 1
// *** IR Dump After Pattern Application ***
func.func @test2(%arg0: i1) -> i32 {
  %false = arith.constant false
  %true = arith.constant true
  %true_0 = arith.constant true
  %0 = arith.xori %arg0, %true_0 : i1
  %1 = arith.andi %arg0, %false : i1
  %2 = arith.andi %0, %true : i1
  %3 = arith.ori %1, %2 : i1
  %4 = arith.extui %3 : i1 to i32
  return %4 : i32
}
```

Reply:
Nice! LGTM